src/antsibull_core/yaml.py

```diff
@@ -43,11 +43,12 @@ def load_yaml_file(path: StrOrBytesPath) -> t.Any:
         return yaml.load(stream, Loader=_SafeLoader)
 
 
-def store_yaml_file(path: StrOrBytesPath, content: t.Any) -> None:
+def store_yaml_file(path: StrOrBytesPath, content: t.Any, append: bool = False) -> None:
```

Comment:
Hmm, I don't really like the approach of writing a file and then appending later on, especially if it happens in the same calling function. Code really should open a file for writing once and then write all content it wants to write instead.

What do you think of the following approaches:
1. Have a function that stores YAML into `bytes`;
2. Have a function that writes YAML into a files object;
3. Or both?

Reply:
Yeah, you're exactly right. I've added a `store_yaml_stream` function to write YAML into a file object.